### ISSUE TITLE
How do you use rosdep

### ISSUE THREAD

**Issue by VoreckLukas** [lopsided98/nix-ros-overlay#392]
Im following a guide to install the [carla ros bridge](https://github.com/carla-simulator/ros-bridge?tab=readme-ov-file) and it tells me to use rosdep. I installed rosdep via pip (couldnt find it in the overlay) but im hesitant to run rosdep init as that would put something in my etc directory and it feels like an anti pattern to using nix

**Comment by lopsided98** [comment]
You are correct that you normally won't use rosdep with Nix (although it is packaged as `python3Packages.rosdep`, since superflore uses it to generate the packages).

For packages that aren't published and automatically included in this overlay (which includes all the carla packages except `carla_msgs`), you have to manually go through the `package.xml` and add each dependency to your Nix shell.
